go/src/koding/kites/kloud/provider/google/stack.go

```diff
@@ -254,6 +257,10 @@ func (s *Stack) ApplyTemplate(c *stack.Credential) (*stack.Template, error) {
 
 	t.Resource["google_compute_instance"] = resource.GCInstance
 
+	t.Variable["koding_stack_id"] = map[string]interface{}{
```

Comment:
I don't at this point we should go into exporting `koding_stack_id` to the user of the template, instead we should just generate unique name for the instance if not set explicitly.

When we're done with improving kloud, then we can improve it for more advanced users by exporting all the needed IDs and adding support for external cloud-init. For now there's no much gain in my opinion in exporting just one ID. I'm fan of full solution, where we design an API first and look back at what we have right now.